util/annotations/annotations.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Annotations is a general wrapper for text that is shown in the
+// Prometheus/Grafana UI along with the query results.
```

Comment:
I wouldn't explicitly call out Grafana here. Maybe be more API centric, along the lines: "... for warnings and other information that is returned by the query API along with the results."

This should also explain that an individual annotation is modeled by a Go `error`, and that `Annotations` deduplicates based on the string returned by `error.Error()`.